Update stacks node api docs (#583)
blockstack/docs.blockstack: _common/core_ref.md

---
layout: core

> title: "Stacks Node API"

Reviewer: I fear this title may be a source of confusion. It can be read as having two meanings:
- "Stacks Node" API
- "Stacks" Node API (as in, a Stacks NodeJS API)

Maybe title it: `stacks-node` API, or Stacks Blockchain API?

Author: Good point. I like `Stacks Blockchain API`, could cover both sidecar and RPC

description: Interacting with the Blockstack Blockchain
permalink: /:collection/:path.html
---

With the launch of Stacks 2.0, a new version of the Blockstack blockchain was released. There are two ways of interacting with the blockchain, either using a RESTful JSON API or by making JSON RPC calls to the blockchain core node.

## JSON RPC API
The Stacks 2.0 blockchain's Rust implementation exposes JSON RPC endpoints, which can be used to interface with blockchain. [You can find the RPC API references here](https://github.com/blockstack/stacks-blockchain/blob/master/docs/rpc-endpoints.md).

## RESTful JSON API

> A RESTful JSON API that is hosted by PBC and has a proxy to the JSON RPC endpoints. [You can find the OpenAPI specification and documentation here](https://blockstack.github.io/stacks-blockchain-sidecar/). You can also make API calls from within the documentation.

Reviewer: ```suggestion
A RESTful JSON API that is hosted by PBC and has a proxy to the JSON RPC endpoints. [You can find the OpenAPI specification and documentation here](https://blockstack.github.io/stacks-blockchain-sidecar/).
```
Redoc doesn't support this out the box. Maybe something we can add later, for the time being we can instruct to use `curl` or `httpie`

> The Explorer API was built to maintain pageable materialized views of the Stacks 2.0 Blockchain. It is a server that exposes a RESTful JSON API, hosted by PBC. It introduces aidditonal functionality (e.g. get all transactions), as well as proxies calls directly to Stacks Node. [You can find the OpenAPI specification and documentation here](https://blockstack.github.io/stacks-blockchain-sidecar/).

Reviewer: ```suggestion
The Explorer API is built to serve pageable materialized views of the Stacks 2.0 Blockchain. It is a server that exposes a RESTful JSON API, hosted by PBC. It introduces additional functionality (e.g. get all transactions), as well as proxies calls directly to Stacks Node. [You can find the OpenAPI specification and documentation here](https://blockstack.github.io/stacks-blockchain-sidecar/).
```

> Using this API (sometimes referred to as "sidecar") requires you to trust the client but provides a faster onboarding experience. It also addresses performance issues for which querying a node itself would be too slow or difficult.

Reviewer: ```suggestion
Using this API (sometimes referred to as "sidecar") requires you to trust the server, but provides a faster onboarding experience. It also addresses performance issues for which querying a node itself would be too slow or difficult.
``` 
server would be more correct terminology no?

---
layout: core
title: "Stacks Node API"
description: Interacting with the Blockstack Blockchain
permalink: /:collection/:path.html
---

> With the launch of Stacks 2.0, a new version of the Blockstack blockchain was released. You currently have two ways of interacting with the blockchain:

Reviewer: I sometimes use [Hemmingway](http://www.hemingwayapp.com/) to point this out, but usually better to write in the active voice: `There are two ways of interacting with the blockchain`.

> - Using a hosted HTTPS REST client that makes RPC calls to the blockchain

Reviewer: I have a slightly different understanding of what a REST client is. I'd consider this to be something like Postman. 

I'd describe the sidecar here: An API we use to address performance issues for which querying a node itself would be too slow/painful. And that, for convenience, it has a proxy to the core node.

- Directly making JSON RPC calls to the blockchain yourself

Using the hosted client requires you to trust the client but provides a faster onboarding.

> ## Hosted REST Client

Reviewer: I think this should be titled as the sidecar (hopefully until we opt to refer to it as the generic stacks-blockchain-api).

> This client communicates with the JSON RPC endpoints. The hosted client references and OpenAPI specification can be find [here](https://blockstack.github.io/stacks-blockchain-sidecar/).

Reviewer: - found*
- What do you mean by hosted client references?

I think I'd refer to this section as API Documentation. Redoc, the api doc generator we're using, has some features to query the API directly would don't think it's valuable to highlight this.

Sidenote, good hyperlink hygiene ensures a link makes sense in isolation of its context. Consider making a larger portion of this text the link, something like: [You can find the OpenAPI specification and documentation here.](6125026d256d44da0aff966ef82189ca30c1e7b1)

Would also add a big API Documentation link at the top of this page.

## JSON RPC Endpoints

> The Stacks 2.0 blockchain's Rust implementation exposes a JSON RPC endpoint, which can be used to interface with blockchain. You can find the references [here](https://github.com/blockstack/stacks-blockchain/blob/master/docs/rpc-endpoints.md).

Reviewer: ```suggestion
The Stacks 2.0 blockchain's Rust implementation exposes JSON RPC endpoints, which can be used to interface with blockchain. You can find the references [here](https://github.com/blockstack/stacks-blockchain/blob/master/docs/rpc-endpoints.md).
```
